tests/mark/check_cluster_use_metadata.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2016 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from ducktape.mark import parametrize, matrix, ignore, MarkedFunctionExpander
+from ducktape.mark.resource import cluster
+
+import pytest
+
+
+class CheckClusterUseAnnotation(object):
+
+    def check_basic_usage_arbitrary_metadata(self):
+        cluster_use_metadata = {
+            'a': 2,
+            'b': 'hi',
+            'num_nodes': 300
+        }
+
+        @cluster(**cluster_use_metadata)
+        def function():
+            return "hi"
+        assert hasattr(function, "marks")
+
+        test_context_list = MarkedFunctionExpander(function=function).expand()
+        assert len(test_context_list) == 1
+        assert test_context_list[0].cluster_use_metadata == cluster_use_metadata
+
+    def check_basic_usage_num_nodes(self):
+        num_nodes = 200
+
+        @cluster(num_nodes=num_nodes)
+        def function():
+            return "hi"
+        assert hasattr(function, "marks")
+
+        test_context_list = MarkedFunctionExpander(function=function).expand()
+        assert len(test_context_list) == 1
+        assert test_context_list[0].expected_num_nodes == num_nodes
+
+    def check_with_parametrize(self):
+        num_nodes = 200
+
+        @cluster(num_nodes=num_nodes)
+        @parametrize(x=1)
+        def f(x, y=2, z=3):
+            return x, y, z
+
+        test_context_list = MarkedFunctionExpander(function=f).expand()
+        assert len(test_context_list) == 1
+        assert test_context_list[0].expected_num_nodes == num_nodes
+
+    def check_beneath_parametrize(self):
+        """ Annotations such as cluster, which add metadata, but do not create new tests, add the metadata to
+        all test cases physically below the annotation.
+
+        In the case of a parametrized test, when @cluster has no parametrize annotations below it,
```

Comment:
Interesting. Not clear to me that we want/need multiple cluster annotations since it seems like it means you're either using variably-sized services or a toggling services on/off since otherwise you'd have a constant # of nodes per test. Variably-sized services seems like the more likely use case to arise (e.g. some sort of scalability test that ramps up # of nodes), but I'm not sure whether we'll run into that in practice or not.

I'm fine with this as is, just seemed unintuitive that it requires specific placement.